js/modules/k6/http/request.go

```diff
@@ -307,19 +308,11 @@ func (c *Client) parseRequest(
 			case "redirects":
 				result.Redirects = null.IntFrom(params.Get(k).ToInteger())
 			case "tags":
-				tagsV := params.Get(k)
-				if goja.IsUndefined(tagsV) || goja.IsNull(tagsV) {
-					continue
-				}
-				tagObj := tagsV.ToObject(rt)
-				if tagObj == nil {
-					continue
-				}
-				tagKeys := tagObj.Keys()
-				result.Tags = make([][2]string, 0, len(tagKeys))
-				for _, key := range tagKeys {
-					result.Tags = append(result.Tags, [2]string{key, tagObj.Get(key).String()})
+				newTags, err := common.ApplyCustomUserTags(rt, result.Tags, params.Get(k))
+				if err != nil {
+					return nil, fmt.Errorf("failed to apply the defined tags: %w", err)
```

Comment:
Same as https://github.com/grafana/k6/pull/2716/files#r990395132, "apply" shouldn't be part of the error because the users are not "applying" anything. Either no wrapping or something like this:
```suggestion
					return nil, fmt.Errorf("invalid HTTP request metric tags: %w", err)
```